<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>

<body>
    <script>
        // 实例化Promise对象
        const p = new Promise(function (resolve, reject) {
            setTimeout(function () {
                // let data = '数据库中的数据';
                resolve('用户数据');
                // let err = '数据读取失败';
                // reject(err);
            }, 1000);
        });

        // 调用promise对象中的then方法，then方法的返回结果是Promise对象，对象状态由回调函数的执行结果决定
        // 如果回调函数中返回的结果是非Promise类型的属性，状态为成功，返回值为对象的成功值 
        // const result = p.then(function (value) {
        //     console.log(value);
        //     // 非Promise类型的对象
        //     // return 123;
        //     // 是Promise对象
        //     // return new Promise((resolve, reject)=>{
        //     //     // resolve('ok');
        //     //     reject('error');
        //     // });
        //     // 抛出错误
        //     throw new Error('出错啦');

        // }, function (reason) { 
        //     console.log(reason);
        // });

        // console.log(result);

        // 链式调用 解决回调地狱
        p.then(value=>{

        }, reason=>{

        }).then(value=>{

        }, reason=>{

        })

    </script>
</body>

</html>